packages/compartment-mapper/src/node-modules.js

```diff
@@ -189,83 +198,83 @@ const findPackage = async (readDescriptor, canonical, directory, name) => {
+const defaultCommonjsLanguageForExtension = /** @type {const} */ ({
   js: 'cjs',
-  ...defaultUncontroversialParsers,
 });
-const defaultModuleParsers = /** @type {const} */ ({
+const defaultModuleLanguageForExtension = /** @type {const} */ ({
   js: 'mjs',
-  ...defaultUncontroversialParsers,
 });
 
 /**
  * @param {object} descriptor
  * @param {string} location
- * @param {object} [options]
- * @param {readonly string[]|string[]} [options.languages]
- * @param {Record<string, string>} [options.uncontroversialParsers]
- * @param {Record<string, string>} [options.commonParsers]
- * @param {Record<string, string>} [options.moduleParsers]
+ * @param {LanguageOptions} languageOptions
  * @returns {Record<string, string>}
  */
-const inferParsers = (
-  descriptor,
-  location,
-  {
-    languages = defaultLanguages,
-    uncontroversialParsers = defaultUncontroversialParsers,
-    commonParsers = defaultCommonParsers,
-    moduleParsers = defaultModuleParsers,
-  } = {},
-) => {
-  const { type, module, parsers } = descriptor;
-  let additionalParsers = Object.create(null);
-  if (parsers !== undefined) {
-    if (typeof parsers !== 'object') {
-      throw Error(
-        `Cannot interpret parser map ${JSON.stringify(
-          parsers,
-        )} of package at ${location}, must be an object mapping file extensions to corresponding languages (mjs for ECMAScript modules, cjs for CommonJS modules, or json for JSON modules`,
-      );
-    }
-    const invalidLanguages = values(parsers).filter(
-      language => !languages.includes(language),
+const inferParsers = (descriptor, location, languageOptions) => {
+  let { moduleLanguageForExtension, commonjsLanguageForExtension } =
+    languageOptions;
+  const {
+    languages,
+    workspaceModuleLanguageForExtension,
+    workspaceCommonjsLanguageForExtension,
+  } = languageOptions;
+
+  // Select languageForExtension options based on whether they are physically
+  // under node_modules, indicating that they have not been built for npm,
+  // so any languages that compile to JavaScript may need additional parsers.
+  if (!location.includes('/node_modules/')) {
+    moduleLanguageForExtension = workspaceModuleLanguageForExtension;
+    commonjsLanguageForExtension = workspaceCommonjsLanguageForExtension;
+  }
+
+  const {
+    type,
+    module,
+    parsers: packageLanguageForExtension = {},
+  } = descriptor;
+
+  // Validate package-local "parsers"
+  if (
+    typeof packageLanguageForExtension !== 'object' ||
+    packageLanguageForExtension === null
+  ) {
+    throw Error(
+      `Cannot interpret parser map ${JSON.stringify(
+        packageLanguageForExtension,
+      )} of package at ${location}, must be an object mapping file extensions to corresponding languages (for example, mjs for ECMAScript modules, cjs for CommonJS modules, or json for JSON modules`,
+    );
+  }
+  const invalidLanguages = values(packageLanguageForExtension).filter(
+    language => !languages.has(language),
```

Comment:
`language` and `languages` look very similar. consider,
```suggestion
    l => !languages.has(l),
```